src/payload.js

```diff
@@ -0,0 +1,28 @@
+const uuid = require('uuid/v1');
+
+function common (config) {
+  return {
+    id: uuid(),
+    timestamp: Date.now(),
+    source: config.source
+  };
+}
+
+module.exports = function formatPayload (config, logs) {
+  if (config.schemaVersion === 2) {
+    return Object.assign(common(config), {
+      'log2amqp-schema-version': '2.1.0',
+      logs: logs.map(l => {
+        const ret = {};
+        ret[l.type] = l.data;
+        return ret;
+      })
+    });
+  } else if (config.schemaVersion === 3) {
+    return Object.assign(common(config), {
+      'log2amqp-schema-version': '3.0.0',
+      logs
```

Comment:
The other possible alternative to this that occurs to me might be something like...
```js
logs.reduce((acc, { type, data }) => ({ ...acc, [type]: [ ...(acc[type] || []), data ] }), {});
```

or if you want the slightly more verbose option...
```js
logs.reduce((acc, { type, data }) => {
  acc[type] = acc[type] || [];
  acc[type].push(data);
  return acc;
}, {});
```